docs/book/src/tasks/experimental-features/cluster-class/change-clusterclass.md

```diff
@@ -1,32 +1,190 @@
 # Changing a ClusterClass
 
-When you change a ClusterClass, the system validates the required changes according to the [compatibility rules defined in the ClusterClass proposal](https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/proposals/202105256-cluster-class-and-managed-topologies.md#clusterclass-compatibility).
+## Selecting a strategy
 
-According to [Cluster API operational practices](https://cluster-api.sigs.k8s.io/tasks/updating-machine-templates.html), the recommended way for updating templates is by template rotation (create a new template, update the template reference in the ClusterClass, and then delete the old template).
+When planning a change to a ClusterClass, users should always take into consideration
+how those changes might impact the existing Clusters already using the ClusterClass, if any.
+
+There are two strategies for defining how a ClusterClass change rolls out to existing Clusters: 
+
+- Roll out ClusterClass changes to existing Cluster in a controlled/incremental fashion.
+- Roll out ClusterClass changes to all the existing Cluster immediately.
+
+The first strategy is the recommended choice for people starting with ClusterClass; it
+requires the users to create a new ClusterClass with the expected changes, and then
+[rebase](#rebase) each Cluster to use the newly created ClusterClass.
+
+By splitting the change to the ClusterClass and its rollout
+to Clusters into separated steps the user will reduce the risk of introducing unexpected
+changes on the existing Clusters, or at least limit the blast radius of those changes
+to a small number of Clusters already rebased (in fact it is similar to a canary deployment).
+
+The second strategy listed above instead requires to change a ClusterClass "in place", which can
+be simpler and faster than creating a new ClusterClass, but this approach
+implies that changes are immediately propagated to all the Clusters already using the 
+ClusterClass being changed; any operation impacting many Clusters has intrinsic risks,
+and it can impact heavily on the underlying infrastructure in case the operation triggers 
+machine rollout across the entire fleet of Clusters.
```

Comment:
nit
```suggestion
The second strategy listed above instead requires changing a ClusterClass "in place", which can
be simpler and faster than creating a new ClusterClass. However this approach
means that changes are immediately propagated to all the Clusters already using the modified
ClusterClass. Any operation impacting many Clusters has intrinsic risks,
and it can impact heavily on the underlying infrastructure in case the operation triggers 
machine rollout across the entire fleet of Clusters.
```